### PR TITLE
Created orders table

### LINKED ISSUE #226: Add filtering for orders

### PR DESCRIPTION
**Pull request by D1nviz**
image

### CODE REVIEW

File: code/src/components/app-table/AppTable.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  AppTableBody,
+  AppTableBox,
+  AppTableContainer,
+  AppTableHead,
+  AppTableRow
+} from "@/components/app-table/components";
+
+import cn from "@/utils/cn/cn";
+
+type classNames = {
```
Comment (mhevyk) [inline code/src/components/app-table/AppTable.tsx]: Types should be in uppercase

File: code/src/components/app-table/AppTable.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  container?: string;
+  head?: string;
+  body?: string;
+};
+
+type AppTableProps<T> = {
```
Comment (mhevyk) [inline code/src/components/app-table/AppTable.tsx]: Make generic more descriptive, for example replace `T` with `BodyItem`

File: code/src/constants/orderStatuses.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
-  IN_PROGRESS: "in progress",
-  SHIPPED: "shipped",
-  DELIVERED: "delivered",
-  CANCELED: "canceled",
-  COMPLETED: "completed"
+  IN_PROGRESS: "In progress",
+  SHIPPED: "Shipped",
+  DELIVERED: "Delivered",
+  CANCELED: "Canceled",
+  COMPLETED: "Completed"
```
Comment (dmtrth25) [inline code/src/constants/orderStatuses.ts]: should be translations for statuses

File: code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.scss [hunk 4]
```diff
@@ -0,0 +1,15 @@
+@import "@design-system";
+
+.spa-order-table__body {
+  @include transition((background-color));
+
+  background-color: var(--spa-order-table-body-bg-color, $white);
```
Comment (dmtrth25) [inline code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.scss:6]: `bg` write full naming for `background-color`

File: code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.scss [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+  &:hover {
+    background-color: var(--spa-order-table-body-hover-bg-color, $light);
```
Comment (dmtrth25) [inline code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.scss:9]: also here

File: code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.scss [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  }
+
+  &-status {
+    width: 75px;
```
Comment (dmtrth25) [inline code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.scss:13]: use `spacing` to define `px` size

File: code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+const OrdersTableBody = ({ order }: OrderTableBodyProps) => {
+  const orderItemStatus = orderStatuses[order.orderStatus];
+  const orderReciever = `${order.receiver.firstName} ${order.receiver.lastName}`;
+
+  const ordersTotalPrice = order.orderItems.reduce(
+    (total, item) => total + item.quantity * item.product.price,
+    0
+  );
```
Comment (dmtrth25) [inline code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.tsx]: keep in mind. Maybe `totalPrice` will be handled from backend side

File: code/src/layouts/tables/orders-table/components/orders-table-head/OrdersTableHead.scss [hunk 8]
```diff
@@ -0,0 +1,6 @@
+@import "@design-system";
+
+.spa-order-table__head {
+  background-color: var(--spa-order-table-head-bg-color, $light-alpha);
```
Comment (dmtrth25) [inline code/src/layouts/tables/orders-table/components/orders-table-head/OrdersTableHead.scss:4]: `bg` -> `background`

File: code/src/constants/orderStatuses.ts (hunk 3 above)
Comment (D1nviz) [reply code/src/constants/orderStatuses.ts:6]: We are using shortcut for background. Look for example AppButton or other app components

Comment (D1nviz) [reply code/src/constants/orderStatuses.ts:6]: We are using shortcut for background. Look for example AppButton or other app components

File: code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.scss (hunk 4 above)
Comment (D1nviz) [reply code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.scss:6]: We are using shortcut for background. Look for example AppButton or other app components

File: code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.scss (hunk 6 above)
Comment (D1nviz) [reply code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.scss:13]: We don't using spacing for width, height properties

File: code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.tsx (hunk 7 above)
Comment (mhevyk) [inline code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.tsx]: I suggest adding some destructuring for `order`

Comment (D1nviz) [reply code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.tsx]: I created the issue and already assigned for refactoring order type. It also includes to remove this reduce functions. But for now it works in this way

File: code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.scss (hunk 4 above)
Comment (dmtrth25) [reply code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.scss:6]: > We are using shortcut for background. Look for example AppButton or other app components

It was implemented by you
image

Follow single approach with your naming creating before

**Review by OliaHrytsak — APPROVED** [review]
Excellent job!

File: code/src/constants/orderStatuses.ts (hunk 3 above)
Comment (D1nviz) [reply code/src/constants/orderStatuses.ts]: added

File: code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.scss (hunk 4 above)
Comment (D1nviz) [reply code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.scss:6]: changed naming of the table-container

File: code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.tsx (hunk 7 above)
Comment (D1nviz) [reply code/src/layouts/tables/orders-table/components/orders-table-body/OrdersTableBody.tsx]: Added

**Review by dmtrth25 — APPROVED** [review]

**Review by mhevyk — APPROVED** [review]

**Review by mhevyk — APPROVED** [review]
Huge work performed! Super!

**Comment by D1nviz** [comment]
Thank you guys for the beautiful comments )

**Pull request merged**